changelog/8014.bugfix.md

```diff
@@ -0,0 +1,15 @@
+Setting `model_confidence=cosine` in `DIETClassifier`, `ResponseSelector` and `TEDPolicy` is deprecated and will no longer be available. This was introduced in Rasa Open Source version `2.3.0` but post-release experiments suggest that using cosine similarity as model's confidences can change the ranking of predicted labels which is wrong.
+
+Also, `model_confidence=inner` is deprecated and is replaced by `model_confidence=linear_norm` as the former produced an unbounded range of confidences which broke the logic of assistants in various other places.
+
+Users are now encouraged to try `model_confidence=linear_norm` which will produce a linearly normalized version of dot product similarities with each value in the range `[0,1]`. This can be done as:
+```yaml
+- name: DIETClassifier
+  model_confidence: linear_norm
+  constrain_similarities: True
+```
+This should ease up tuning fallback thresholds as confidences for wrong predictions are better distributed across the range `[0, 1]`.
```

Comment:
please mention the fallback classifier here and link to it

Reply:
Linking it to the dedicated page on fallback handoff as it covers both NLU and Core fallback